data_structures/src/superblock.rs

```diff
@@ -499,27 +506,49 @@ pub fn calculate_superblock_signing_committee(
     signing_committee_size: u32,
     current_superblock_index: u32,
     superblock_hash: Hash,
+    block_epoch: Epoch,
 ) -> HashSet<PublicKeyHash> {
     // If the number of identities is lower than committee_size all the members of the ARS sign the superblock
     if ars_identities.len() < usize::try_from(signing_committee_size).unwrap() {
         ars_identities.identities
     } else {
         // Hash of the current_index, to avoid potential committee collisions
         let index_hash = Hash::from(calculate_sha256(&current_superblock_index.to_be_bytes()));
-        // Start counting the members of the subset from the superblock_hash plus superblock index hash
-        let mut first = u32::from(*superblock_hash.as_ref().get(0).unwrap())
-            + u32::from(*index_hash.as_ref().get(0).unwrap());
-        // We need to choose a first member from all the potential ARS members
-        first %= ars_identities.len() as u32;
-
-        // Get the subset
-        let subset = magic_partition(
-            &ars_identities.ordered_identities.to_vec(),
-            first.try_into().unwrap(),
-            signing_committee_size.try_into().unwrap(),
-        );
-        let hs: HashSet<PublicKeyHash> = subset.iter().cloned().collect();
-        hs
+        let first_byte_sb_hash = *superblock_hash.as_ref().get(0).unwrap();
+        let first_byte_index_hash = *index_hash.as_ref().get(0).unwrap();
+        let second_byte_index_hash = *index_hash.as_ref().get(1).unwrap();
+
+        if after_second_hard_fork(block_epoch, get_environment()) {
+            // Start counting the members of the subset from:
+            // 'first_byte_sb_hash'x263 + 'first_byte_index_hash'x257 + 'second_byte_index_hash'
+            let mut first = u64::from(first_byte_sb_hash) * 263u64
+                + u64::from(first_byte_index_hash) * 257u64
+                + u64::from(second_byte_index_hash);
+            // We need to choose a first member from all the potential ARS members
+            first %= ars_identities.len() as u64;
```

Comment:
With the new algorithm the value of `first` is not very important. And taking into account the complexity of this logic, it would be nice to remove it if possible.

But if we remove it then the committees will not depend on the superblock hash and I assume that that's required. So I suggest to use `index_hash ^ superblock_hash` as the "rand_distribution", instead of `index_hash`. And hardcode `first` to `0`. What do you think?

Reply:
If the rand_distribution use bytes as a random offsets, the biggest offset would be +255, so if the distance between candidates in the initial distribution without randomness is bigger than that there would be some positions that never will be chosen to participate in the committee